### PR TITLE
Canvas location lock improvement.

### PR DESCRIPTION
**Pull request by mohsenD98**
### PR Description:
At present, when the canvas location is locked and the user zooms in or out with a mouse, the lock will break.

### Fix: 
Using same approach in `zoomInButton` and `zoomOutButton`:
```QML
if (gnssButton.followActive)
        gnssButton.followActiveSkipExtentChanged = true;
```

### Demo: 
Now zooming in/out wont break lock.   

[Screencast from 2024-10-19 16-49-36.webm](https://github.com/user-attachments/assets/dffa4b78-986c-4f3c-8344-fe48dd094245)

### CODE REVIEW

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for 30ba94e62780937ee6d0207724e6cd7040e1ced9: [**arm64-android**](https://qfieldapks.sos-ch-dk-2.exoscale-cdn.com/ci-builds/qfield-dev-5749-3ec90fafa6324d243ed123ed30524fae6fe11c44-android-arm64.apk)
<details>
<summary>Other architectures</summary>

- [arm-neon-android](https://qfieldapks.sos-ch-dk-2.exoscale-cdn.com/ci-builds/qfield-dev-5749-3ec90fafa6324d243ed123ed30524fae6fe11c44-android-arm-neon.apk)
- [x64-android](https://qfieldapks.sos-ch-dk-2.exoscale-cdn.com/ci-builds/qfield-dev-5749-3ec90fafa6324d243ed123ed30524fae6fe11c44-android-x64.apk)
- [x86-android](https://qfieldapks.sos-ch-dk-2.exoscale-cdn.com/ci-builds/qfield-dev-5749-3ec90fafa6324d243ed123ed30524fae6fe11c44-android-x86.apk)
- [**all access arm64-android**](https://qfieldapks.sos-ch-dk-2.exoscale-cdn.com/ci-builds/qfield_all_access-dev-5749-3ec90fafa6324d243ed123ed30524fae6fe11c44-android-arm64.apk)
- [all access arm-neon-android](https://qfieldapks.sos-ch-dk-2.exoscale-cdn.com/ci-builds/qfield_all_access-dev-5749-3ec90fafa6324d243ed123ed30524fae6fe11c44-android-arm-neon.apk)
- [all access x64-android](https://qfieldapks.sos-ch-dk-2.exoscale-cdn.com/ci-builds/qfield_all_access-dev-5749-3ec90fafa6324d243ed123ed30524fae6fe11c44-android-x64.apk)
- [all access x86-android](https://qfieldapks.sos-ch-dk-2.exoscale-cdn.com/ci-builds/qfield_all_access-dev-5749-3ec90fafa6324d243ed123ed30524fae6fe11c44-android-x86.apk)
</details>

**Comment by nirvn** [comment]
@mohsenD98 , we need to change this a bit so we don't introduce a "random" item id within MapCanvas.qml (makes it a lot harder to test individual components).

I'd go for adding a signal aboutToZoom(), and then add the logic in qgismobileapp.qml (where gnssButton lives) by listening for that signal.

**Comment by mohsenD98** [comment]
> @mohsenD98 , we need to change this a bit so we don't introduce a "random" item id within MapCanvas.qml (makes it a lot harder to test individual components).
> 
> I'd go for adding a signal aboutToZoom(), and then add the logic in qgismobileapp.qml (where gnssButton lives) by listening for that signal.

Good point :+1:

File: src/qml/MapCanvas.qml [hunk 1]
```diff
@@ -51,6 +51,9 @@ Item {
   //! Emitted when a release happens after a long press
   signal longPressReleased(var type)
 
+  //! Emitted when a zoom action is about to occur, allowing for pre-zoom adjustments
+  signal aboutToZoom
```
[the remark this answers is not in the document]
Comment (mohsenD98) [reply src/qml/MapCanvas.qml]: Ah, right  :+1:

**Review by nirvn — APPROVED** [review]

**Pull request merged**